be/src/storage/lake/transactions.cpp

```diff
@@ -172,14 +172,13 @@ StatusOr<TabletMetadataPtr> publish_version(TabletManager* tablet_mgr, int64_t t
             << ", new_version: " << new_version;
 
     auto commit_time = txns.back().commit_time();
-    auto new_metadata_path = tablet_mgr->tablet_metadata_location(tablet_id, new_version);
-    auto cached_new_metadata = tablet_mgr->metacache()->lookup_tablet_metadata(new_metadata_path);
-    if (cached_new_metadata != nullptr) {
-        // The retries may be caused by some tablets failing to publish in a partition
-        // set the following log as debug log to prevent excessive logging
-        VLOG(1) << "Skipped publish version because target metadata found in cache. tablet_id=" << tablet_id
+    // protobuf can not guarantee idempotent, so check new tablet meta here to prevent
+    // same tablet meta from being written twice
+    auto res = tablet_mgr->get_tablet_metadata(tablet_id, new_version);
```

Comment:
this doesn't prevent the access to cached metadata after publish success and potentially lead to data inconsistency, should the metadata be cleared from the meta cache, or check if the metadata file size matches the one cached in memory?

Reply:
get_tablet_metadata will check meta cache first, so it's fine, if meta cache does exist, it will check starlet